services/explorer/src/components/Recent.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useEffect, useState } from "react";
+import {
+    Box,
+    Button,
+    IconButton,
+    Typography,
+    Select,
+    MenuItem,
+    FormControl
+} from "@mui/material";
+import { GatekeeperEvent, GatekeeperInterface } from "@mdip/gatekeeper/types";
+import { getTypeStyle, handleCopyDID } from "../shared/utilities.js";
+import ContentCopy from "@mui/icons-material/ContentCopy";
+
+const networksEnv = import.meta.env.VITE_OPERATION_NETWORKS || "hyperswarm";
+const knownRegistries: string[] = ["All", ...networksEnv.split(",").map((n: string) => n.trim())];
+if (!knownRegistries.includes("local")) {
+    knownRegistries.push("local");
+}
+
+function Recent(
+    {
+        gatekeeper,
+        setError,
+        onDidClick
+    }: {
+        gatekeeper: GatekeeperInterface;
+        setError: (error: any) => void;
+        onDidClick: (did: string) => void;
+    }) {
+    const [events, setEvents] = useState<GatekeeperEvent[]>([]);
+    const [total, setTotal] = useState<number>(0);
+    const [eventCount, setEventCount] = useState<number>(50);
+    const [page, setPage] = useState<number>(0);
+    const [registry, setRegistry] = useState<string>("All");
+
+    useEffect(() => {
+        let isMounted = true;
+        let intervalId: NodeJS.Timeout | undefined;
+
+        async function fetchRecent() {
+            try {
+                const dids = (await gatekeeper.getDIDs()) as string[];
+                let allEvents = (await gatekeeper.exportDIDs(dids)).flat();
```

Comment:
Just fyi, gatekeeper.exportDIDs() is equivalent to the above.

Reply:
OTOH for a "recent" view, you may only want to export DIDs that were updated recently (like in the last N days where N is not large)?